libcontainer/cgroups/systemd/unified_hierarchy.go

```diff
@@ -197,14 +189,26 @@ func (m *UnifiedManager) GetUnifiedPath() (string, error) {
 	}
 	return unifiedPath, nil
 }
+
+func getv2Path(c *configs.Cgroup) (string, error) {
+	slice := "system.slice"
+	if c.Parent != "" {
+		slice = c.Parent
+	}
+
+	slice, err := ExpandSlice(slice)
+	if err != nil {
+		return "", err
+	}
+
+	return filepath.Join(UnifiedMountpoint, slice, getUnitName(c)), nil
+}
+
 func createCgroupsv2Path(path string) (Err error) {
 	content, err := ioutil.ReadFile("/sys/fs/cgroup/cgroup.controllers")
 	if err != nil {
 		return err
 	}
-	if !filepath.HasPrefix(path, "/sys/fs/cgroup") {
-		return fmt.Errorf("invalid cgroup path %s", path)
-	}
```

Comment:
Why remove?

Reply:
We're calling this function from a single place in the same file with path from `getv2Path()` which clearly sets prefix to `/sys/fs/cgroup/`.

I can reinstate the check but it seems redundant now.